server/src/main/java/ch/ethz/lapis/api/SampleService.java

```diff
@@ -326,56 +324,76 @@ public List<InsertionStore.InsertionCount> getInsertions(
         }
     }
 
-
-    public void getFasta(
+    public void getNucSequencesInFastaFormat(
         BaseSampleRequest request,
         boolean aligned,
         OrderAndLimitConfig orderAndLimit,
         OutputStream outputStream
     ) {
-        // Filter
-        List<Integer> ids = new QueryEngine().filterIds(Database.getOrLoadInstance(dbPool), request);
-        if (ids.isEmpty()) {
+        var filteredIds = new QueryEngine().filterIds(Database.getOrLoadInstance(dbPool), request);
+        if (filteredIds.isEmpty()) {
             return;
         }
 
-        // Filter further by the other metadata and prepare the response
+        runInDatabaseContextWithoutAutoCommit(ctx -> {
+            var metaTable = YMainMetadata.Y_MAIN_METADATA;
+            var sequenceTable = YMainSequence.Y_MAIN_SEQUENCE;
+
+            var sequenceColumn = aligned ? sequenceTable.SEQ_ALIGNED_COMPRESSED : sequenceTable.SEQ_ORIGINAL_COMPRESSED;
+
+            var idsTable = getIdsTable(filteredIds, ctx);
+            var sequenceIdentifierColumn = LapisMain.globalConfig.getApiOpennessLevel() == OpennessLevel.OPEN
+                ? metaTable.GENBANK_ACCESSION
+                : metaTable.GISAID_EPI_ISL;
+
+            var statement = ctx
+                .select(sequenceIdentifierColumn, sequenceColumn)
+                .from(
+                    idsTable
+                        .join(metaTable).on(idsTable.field("id", Integer.class).eq(metaTable.ID))
+                        .join(sequenceTable).on(metaTable.ID.eq(sequenceTable.ID))
+                );
+
+            streamFastaSequences(orderAndLimit, outputStream, sequenceColumn, sequenceIdentifierColumn, statement);
+        });
+    }
+
+    public StreamingResponseBody getAaSequencesInFastaFormatStream(
+        BaseSampleRequest request,
+        OrderAndLimitConfig orderAndLimit,
+        String gene
+    ) {
+        var filteredIds = new QueryEngine().filterIds(Database.getOrLoadInstance(dbPool), request);
+        if (filteredIds.isEmpty()) {
+            return outputStream -> {
+            };
+        }
+        return outputStream -> runInDatabaseContextWithoutAutoCommit(ctx -> {
+            var metaDataTable = YMainMetadata.Y_MAIN_METADATA;
+            var aaSequenceTable = YMainAaSequence.Y_MAIN_AA_SEQUENCE;
+
+            var sequenceIdentifierColumn = metaDataTable.GENBANK_ACCESSION;
```

Comment:
```suggestion
            var sequenceIdentifierColumn = LapisMain.globalConfig.getApiOpennessLevel() == OpennessLevel.OPEN
                ? metaTable.GENBANK_ACCESSION
                : metaTable.GISAID_EPI_ISL;
```

Similar to the above, the identifier column is different for the two different data sources.